services/catalog/src/simcore_service_catalog/db/repositories/services.py

```diff
@@ -173,30 +237,37 @@ async def get_service_access_rights(
         self,
         key: str,
         version: str,
-        product_name: str,
+        product_name: Optional[str] = None,
     ) -> List[ServiceAccessRightsAtDB]:
+        """
+        - If product_name is not specificed, then all are considered in the query
```

Comment:
what is a use-case for such a thing? is that not a bit weird to ask for access rights whatever the product_name?

Reply:
Given a service, the new auto-update policy needs access rights for that service's previous version including *all* products

Reply:
ok, makes sense in that case.